pac4j-jwt/src/main/java/org/pac4j/jwt/profile/JwtGenerator.java

```diff
@@ -38,27 +46,28 @@
 
     protected final Logger logger = LoggerFactory.getLogger(getClass());
 
-    private final String secret;
-    private boolean encrypted = true;
+    private final String signingSecret;
+    private final String encryptionSecret;
 
     /**
-     * Initializes the generator that will create JWT tokens that are both signed and encrypted.
+     * Initializes the generator that will create JWT tokens that are signed but not encrypted.
      *
-     * @param secret The secret. Must be at least 256 bits long and not {@code null}
+     * @param signingSecret The signingSecret. Must be at least 256 bits long and not {@code null}
      */
-    public JwtGenerator(String secret) {
-        this.secret = secret;
+    public JwtGenerator(final String signingSecret) {
+        this(signingSecret, null);
```

Comment:
OK. I see why you use `null` to replace `encrypted`, but this generates a problem: if someone upgrades to pac4j v1.8.2, data will stop being encrypted when keeping the same constructor (the one with only one parameter).
So we must keep the current constructors:

```
public JwtGenerator(final String secret) {
   this(secret, true);
}
public JwtGenerator(final String secret, final boolean encrypted) {
  this.signingSecret = secret;
  if (encrypted) {
    this.encryptionSecret = secret;
  }
}
```

and add a new one:

```
public JwtGenerator(final String signingSecret, final String encryptionSecret) {
  this.signingSecret = signingSecret;
  this.encryptionSecret = encryptionSecret;
}
```